src/pages/SignUp.js

```diff
@@ -75,77 +104,185 @@ export default function SignUp() {
+                    autoComplete="family-name"
+                    inputProps={{ pattern: "[a-zA-Z0-9]*" }} // Allow alphanumeric characters only
+                    value={lastname}
+                    onChange={(event, data) => { setLastName(event.target.value) }}
+                  />
+                </Grid>
+
+                {/* Email address */}
+                <Grid item xs={12} >
+                  <TextField
+                    required
+                    fullWidth
+                    id="email"
+                    label="Email Address"
+                    name="email"
+                    autoComplete="email"
+                    value={email}
+                    helperText="Valid hawk or gmail id"
+                    inputProps={{ pattern: "^[a-zA-Z0-9]+@(hawk.iit.edu|gmail.com)$" }}
+                    onChange={(event, data) => { setEmail(event.target.value) }}
+                  />
+                </Grid>
+
+                {/* Gender */}
+                <Grid item xs={12} sm={6}>
+                  <FormControl sx={{ m: 1, minWidth: 100 }}>
+                    <InputLabel id="gender-select-autowidth-label">Gender</InputLabel>
+                    <Select
+                      labelId="gender-label"
+                      id="gender"
+                      name='gender'
+                      onChange={(event, data) => { setGender(event.target.value) }}
+                      label="Gender"
+                      fullWidth
+                      value={gender}
+                      required
+                    >
+                      <MenuItem value=""><em>None</em></MenuItem>
+                      <MenuItem value={"Male"}>Male</MenuItem>
+                      <MenuItem value={"Female"}>Female</MenuItem>
+                    </Select>
+                  </FormControl>
+                </Grid>
+
+                {/* User Type */}
+                <Grid item xs={12} sm={6}>
+                  <FormControl sx={{ m: 1, minWidth: 100 }}>
+                    <InputLabel id="user-select-autowidth-label">User Type</InputLabel>
+                    <Select
+                      labelId="user-label"
+                      id="user"
+                      name='user'
+                      value={user}
+                      onChange={(event, data) => { setUserType(event.target.value) }}
+                      label="User Type"
+                      fullWidth
+                      required
+                    >
+                      <MenuItem value=""><em>None</em></MenuItem>
+                      <MenuItem value={"User"}>User</MenuItem>
+                      <MenuItem value={"Event Provider"}>Event Provider</MenuItem>
+                      <MenuItem value={"System Admin"}>System Admin</MenuItem>
+                      <MenuItem value={"Content Moderator"}>Content Moderator</MenuItem>
+                    </Select>
+                  </FormControl>
+                </Grid>
+
+                {/* Age */}
```

Comment:
We need to allow the value of Age to be 50+ as well. Probably prof is 50+. (lets keep it as text with input validation like : Age cant be greater than 110 or less than 5)